File: src/main/java/lib/commands/CommandExtensions.java
package lib.commands;

import edu.wpi.first.wpilibj2.command.WrapperCommand;
import edu.wpi.first.wpilibj2.command.Command;
import java.util.function.BooleanSupplier;

/** Command helpers replacing the Kotlin extension utilities. */
public final class CommandExtensions {
  private CommandExtensions() {}

  public static WrapperCommand withIsFinished(Command command, BooleanSupplier isFinished) {
    return new WrapperCommand(command) {
      @Override
      public boolean isFinished() {
        return isFinished.getAsBoolean();
      }
    };
  }
}

File: src/main/java/lib/commands/CommandExtensions.kt
package lib.commands

import edu.wpi.first.wpilibj2.command.Command
import edu.wpi.first.wpilibj2.command.WrapperCommand
import java.util.function.BooleanSupplier

fun Command.withIsFinished(isFinished: BooleanSupplier): WrapperCommand {
    return object : WrapperCommand(this) {
        override fun isFinished(): Boolean {
            return isFinished.asBoolean
        }
    }
}